pages/index.vue

```diff
@@ -1,19 +1,94 @@
 <script setup lang="ts">
-import { useQuery } from '@vue/apollo-composable'
-import { GET_CHARACTERS } from '@/api/queries'
-import { ElTable, ElTableColumn, ElContainer } from 'element-plus'
+import { ElContainer, ElImage, ElPagination, ElCard } from "element-plus";
+import { useCharacterList } from "@/composables/useCharactersList";
+import BaseLoader from "@/base/BaseLoader";
+const currentPage = ref(1);
+const { result, loading } = useCharacterList({ page: currentPage });
 
-
-const { loading, result } = useQuery(GET_CHARACTERS)
 </script>
+
 <template>
   <el-container>
-    <div v-if="result">
-      <el-table :data="result.characters.results" style="width: 100%">
-        <el-table-column prop="name" label="Name" width="180" />
-        <el-table-column prop="id" label="Id" width="180" />
-      </el-table>
+    <div v-if="result" class="characters__wrap">
+      <div class="characters">
+        <nuxt-link
+          v-for="character in result.characters.results"
+          :key="character.id"
+          class="characters-link"
+          :to="{ path: `/character-detail/${character.id}` }"
+        >
+          <el-card
+            shadow="hover"
+            class="characters-card"
+            :body-style="{ padding: '0px' }"
+          >
+            <el-image :src="character.image" />
+            <div
+              class="characters-desc"
+              :class="`characters-desc__${character.gender.toLowerCase()}`"
+              style="padding: 14px"
+            >
+              <div>
+                <b>Name: </b><span>{{ character.name }}</span>
+              </div>
+              <div>
+                <b>Location: </b><span v-if="character.location">{{
+                  character.location.name
+                }}</span>
+              </div>
+            </div>
+          </el-card>
+        </nuxt-link>
+      </div>
+      <el-pagination
+        v-model:currentPage="currentPage"
+        :total="result.characters.info.pages"
+      />
     </div>
-    <div v-if="loading">Loading...</div>
+    <base-loader v-if="loading" />
   </el-container>
 </template>
+
+<style lang="scss">
+.characters {
```

Comment:
I'm not a fan of custom CSS for each page.
Try to create reusable components for that. Also, don't forget to use WindiCSS, you really don't need to write all these custom styles